### PR TITLE
Certman Bug Fixes and Helper Utility

### PR DESCRIPTION
**Pull request by bbengfort**
### Scope of changes

We had an issue with a TRISA TestNet registration today where Sectigo rejected the domain name causing multiple critical alerts to trigger. To remediate the situation we put the registration back to "PENDING_REVIEW". This PR contains the helper script in the `reissuer` command that allowed us to do that.

While working on that, I noticed that the batch names had an extra `")"` at the end, so I fixed that minor typo, and since I was there, I also fixed the typo where Certman saves the VASP every day even if it hasn't been changed (causing the last updated timestamp to be unusable). 

Fixes SC-10097
Fixes SC-10325

### Type of change

- [x] bug fix
- [ ] new feature
- [ ] documentation
- [ ] other (describe)

### Acceptance criteria

For the most part you can ignore the helper command in `reissuer` -- we've already run this command and it worked. The real question is about certman. Is the `updateRequired` bool sufficient to handle the switch with all the various `continue` statements? Is there a better way we could implement it and still have effective error handling? Is this maintainable in the future? 

### Author checklist

- [ ] I have manually tested the change and/or added automation in the form of unit tests or integration tests
- [ ]  I have updated the dependencies list
- [ ]  I have recompiled and included new protocol buffers to reflect changes I made
- [ ]  I have added new test fixtures as needed to support added tests
- [ ]   Check this box if a reviewer can merge this pull request after approval (leave it unchecked if you want to do it yourself)
- [ ]  I have moved the associated Shortcut story to "Ready for Review"

### Reviewer(s) checklist

- [ ] Any new user-facing content that has been added for this PR has been QA'ed to ensure correct grammar, spelling, and understandability.

### CODE REVIEW

File: pkg/gds/certman/certs.go [hunk 1]
```diff
@@ -779,19 +782,30 @@ vaspsLoop:
...
 			}
+			updateRequired = true
 
 		// Thirty days before expiration, send the reissuance reminder to the VASP and the TRISA admin.
 		case daysBeforeExpiration <= 30:
+			// If the reminder fails do not stop processing and attempt to send reminder to admins
 			if err = c.email.SendContactReissuanceReminder(vasp, 30, reissuanceDate); err != nil {
 				log.Error().Err(err).Str("vasp_id", vasp.Id).Msg("error sending thirty day reissuance reminder")
+			} else {
+				updateRequired = true
```
Comment (pdeziel) [inline pkg/gds/certman/certs.go:793]: I'm not sure this is correct, we have some internal logic in this method which means we may or may not send the email, and therefore the log is not necessarily always updated.

**Review by pdeziel — COMMENTED** [review]
For the update VASP issue, maybe a more maintainable solution is to snapshot the VASP object (or just the email log) at the beginning of the loop and then utilize the `Equal` from the protobuf package to determine if we've made any changes?

The rest of the PR looks good, we just need to figure out how to do the `updateRequired` check because as described below I don't think the current solution is necessarily correct.

File: pkg/gds/certman/certs.go (hunk 1 above)
Comment (bbengfort) [reply pkg/gds/certman/certs.go:793]: Ah ok - glad you reviewed this PR! I'll see if I can get in the proto.Equal check.

**Comment by bbengfort** [comment]
@pdeziel this is ready for review again - I've used a hash based mechanism to test equality with the original proto data. I wrote a check for the signature method, but I haven't tested if this will correctly update the VASP during certman iterations. Let me know what you think.

File: pkg/models/v1/models.go [hunk 2]
```diff
@@ -601,3 +603,16 @@ func ValidateVASP(vasp *pb.VASP, partial bool) (err error) {
...
+// equality or to detect if a VASP record has been changed.
+func VASPSignature(vasp *pb.VASP) (_ []byte, err error) {
+	var data []byte
+	if data, err = proto.Marshal(vasp); err != nil {
+		return nil, err
+	}
+
+	sum := sha512.Sum512(data)
+	return sum[:], nil
+}
```
[the remark this answers is not in the document]
Comment (bbengfort) [reply pkg/models/v1/models.go:618]: `proto.Equal` would require us to keep two pointers of the VASP in memory (the before and after pointer) and to be careful that we didn't modify the before pointer by accident (and this seemed a little fragile). To get a fixed representation at a point in time, we'd have to do what proto.Equal does - marshal the protocol buffers to bytes - however, for equality, we don't really need all the bytes and VASP messages can be large, so the hash simplifies things a bit.

File: pkg/gds/certman/certs.go [hunk 3]
```diff
@@ -782,16 +790,32 @@ vaspsLoop:
...
-		if err = c.db.UpdateVASP(vasp); err != nil {
-			log.Error().Err(err).Str("vasp_id", vasp.Id).Msg("error updating the vasp record in the database")
+
+		// Perform VASP hash check to determine if the VASP has been updated, and if it
+		// has, save the updates back to the database (otherwise do not save so that the
+		// last modified timestamp is not updated every day).
+		var updated []byte
+		if updated, err = models.VASPSignature(vasp); err != nil {
+			log.Error().Err(err).Str("vasp_id", vasp.Id).Msg("could not compute signature for VASP")
+			continue vaspsLoop
```
[the remark this answers is not in the document]
Comment (bbengfort) [reply pkg/gds/certman/certs.go:810]: Yes, if it fails the next day the email will be sent again (and hopefully it won't fail then) -- that's also true if we fail to save the record to the database.

**Comment by bbengfort** [comment]
@pdeziel before we deploy this, let's make sure we test in staging!

**Pull request merged**